### PR TITLE
Pack publish date

### PR DESCRIPTION
**Pull request by barchen1**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/31105

## Description
Fixed an issue where pack's created date was the date when the developer created it and not the pack's publish date.

## Screenshot
Screen Shot 2020-12-27 at 12 26 38

Screen Shot 2020-12-27 at 12 26 20

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -1421,6 +1438,26 @@ def format_metadata(self, user_metadata, pack_content_items, integration_images,
...
+        # load changelog from downloaded index
+        changelog_index_path = os.path.join(index_folder_path, self._pack_name, Pack.CHANGELOG_JSON)
+        changelog = {}
+        if os.path.exists(changelog_index_path):
+            try:
+                with open(changelog_index_path, "r") as changelog_file:
+                    changelog = json.load(changelog_file)
+            except json.JSONDecodeError:
+                changelog = {}
+        return changelog.get(Pack.PACK_INITIAL_VERSION, {}).get('released', datetime.utcnow().
```
Comment (guykeller) [inline Tests/Marketplace/marketplace_services.py:1458]: Let's split this into two lines, even three, for better readability

File: Tests/secrets_white_list.json [hunk 2]
```diff
@@ -2672,7 +2672,10 @@
     "vulnInfoList",
     "NAS-Identifier=de-wilm-251littl-idf3b-s2",
     "NAS-Port-Id=GigabitEthernet2/0/05",
-    "--memory=1g##--memory-swap=-1##--pids-limit=256##--ulimit=nofile=1024:8192"
+    "--memory=1g##--memory-swap=-1##--pids-limit=256##--ulimit=nofile=1024:8192",
+    "oproxy-dev.shared_views.top_packs",
+    "Pack.EXCLUDE_DIRECTORIES",
+    "Author_image.png"
```
Comment (guykeller) [inline Tests/secrets_white_list.json:2678]: Why are these added?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Tests/secrets_white_list.json (hunk 2 above)
Comment (barchen1) [reply Tests/secrets_white_list.json]: Cause `demisto-sdk secrets` failed, but now it failed on other stuff

File: Tests/Marketplace/marketplace_services.py [hunk 3]
```diff
@@ -647,15 +653,16 @@ def _get_downloads_count(self, packs_statistic_df):
 
         return downloads_count
 
-    @staticmethod
-    def _create_changelog_entry(release_notes, version_display_name, build_number, new_version=True):
+    def _create_changelog_entry(self, release_notes, version_display_name, build_number, new_version=True,
+                                initial_release=False):
```
[the remark this answers is not in the document]
Comment (guykeller) [reply Tests/Marketplace/marketplace_services.py:657]: You did not change it everywhere yet

File: Tests/Marketplace/marketplace_services.py [hunk 4]
```diff
@@ -1421,6 +1435,29 @@ def format_metadata(self, user_metadata, pack_content_items, integration_images,
         finally:
             return task_status
 
+    def _handle_pack_create_date(self, index_folder_path):
```
Comment (guykeller) [inline Tests/Marketplace/marketplace_services.py:1438]: If it gets the part creation date, why call it `handle`? What knowledge does the name `handle_pack_create_date` provide to the reader?

File: Tests/Marketplace/marketplace_services.py [hunk 5]
```diff
@@ -1421,6 +1435,29 @@ def format_metadata(self, user_metadata, pack_content_items, integration_images,
...
+
+        # load changelog from downloaded index
+        changelog_index_path = os.path.join(index_folder_path, self._pack_name, Pack.CHANGELOG_JSON)
+        changelog = {}
+        if os.path.exists(changelog_index_path):
+            try:
+                with open(changelog_index_path, "r") as changelog_file:
+                    changelog = json.load(changelog_file)
+            except json.JSONDecodeError:
+                changelog = {}
```
Comment (guykeller) [inline Tests/Marketplace/marketplace_services.py:1455]: When would this happen?

File: Tests/Marketplace/marketplace_services.py [hunk 6]
```diff
@@ -1421,6 +1435,29 @@ def format_metadata(self, user_metadata, pack_content_items, integration_images,
...
+        init_changelog_version = changelog.get(Pack.PACK_INITIAL_VERSION, {})
```
Comment (guykeller) [inline Tests/Marketplace/marketplace_services.py:1456]: Does this name truly represent what's in this variable?

File: Tests/secrets_white_list.json [hunk 7]
```diff
@@ -2674,7 +2674,10 @@
     "vulnInfoList",
     "NAS-Identifier=de-wilm-251littl-idf3b-s2",
     "NAS-Port-Id=GigabitEthernet2/0/05",
-    "--memory=1g##--memory-swap=-1##--pids-limit=256##--ulimit=nofile=1024:8192"
+    "--memory=1g##--memory-swap=-1##--pids-limit=256##--ulimit=nofile=1024:8192",
+    "@pytest.mark.parametrize",
+    "TestGetSuccessfulAndFailedPacks",
+    "user_pack_metadata.json"
```
Comment (guykeller) [inline Tests/secrets_white_list.json:2680]: why were these added?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Tests/Marketplace/marketplace_services.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (barchen1) [reply Tests/Marketplace/marketplace_services.py:657]: done

File: Tests/Marketplace/marketplace_services.py (hunk 4 above)
Comment (barchen1) [reply Tests/Marketplace/marketplace_services.py]: changed

File: Tests/Marketplace/marketplace_services.py (hunk 5 above)
Comment (barchen1) [reply Tests/Marketplace/marketplace_services.py:1455]: If `json.load` failed to load the changelog file we will get `datetime.utcnow()` as pack creation date

File: Tests/Marketplace/marketplace_services.py (hunk 6 above)
Comment (barchen1) [reply Tests/Marketplace/marketplace_services.py]: I think it is, this var is for the first version in the changelog file ( 1.0.0)
changed to `initial_changelog_version`

File: Tests/secrets_white_list.json (hunk 7 above)
Comment (barchen1) [reply Tests/secrets_white_list.json:2680]: I ran `demisto-sdk secrets` and I got those strings as secrets

**Review by guykeller — APPROVED** [review]

**Review by guykeller — APPROVED** [review]
